#### Практическое задание: Урок 2. Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

##### 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Попадение в цель - дискретная величина, поэтому для расчета можно использовать формулу Бернулли.

In [1]:
from math import factorial

In [2]:
n = 100
k = 85
p = 0.8
q = 0.2

Найдем количество сочетаний (С):

In [3]:
C = int(factorial(n) / (factorial(k) * factorial(n - k)))

In [4]:
C

253338471349988640

Применим формулу Бернулли для расчета вероятности:

In [5]:
P = C * p**(k) * q**(n-k)

In [6]:
P

0.048061793700746556

##### Ответ: вероятность того, что стрелок попадет в цель ровно 85 раз равна 0.048061793700746556

##### 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

In [7]:
n = 5000
k = 0
p = 0.0004
q = 1 - p

Найдем количество сочетаний (С):

In [8]:
C = int(factorial(n) / (factorial(k) * factorial(n - k)))

In [9]:
C

1

Применим формулу Бернулли для расчета вероятности:

In [10]:
P = C * p**(k) * q**(n-k)

In [11]:
P

0.13528114551440706

2.2 Какова вероятность, что перегорят ровно две?

In [12]:
k = 2

In [13]:
C = int(factorial(n) / (factorial(k) * factorial(n - k)))

In [14]:
C

12497500

In [15]:
P = C * p**(k) * q**(n-k)

In [16]:
P

0.2707247150266753

##### Ответ: 1) вероятность, что ни одна из них не перегорит в первый день равна 0.13528114551440706 2) вероятность, что перегорят ровно две равна 0.2707247150266753

##### 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [17]:
n = 144
k = 70
p = 0.5
q = 0.5

Найдем количество сочетаний (С):

In [18]:
C = int(factorial(n) / (factorial(k) * factorial(n - k)))

In [19]:
C

1400749509083708712353342007350408066891776

Применим формулу Бернулли для расчета вероятности:

In [20]:
P = C * p**(k) * q**(n-k)

In [21]:
P

0.06281178035144776

##### Ответ: вероятность, что орел выпадет ровно 70 раз равна 0.06281178035144776

##### 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

Решение 4.1 Какова вероятность того, что все мячи белые? 

Найдем вероятность, что из первого ящика вытащили оба белых мяча:

In [22]:
P1 = 7/10 * 6/9

In [23]:
P1

0.46666666666666656

Найдем вероятность, что из второго ящика вытащили оба белых мяча:

In [24]:
P2 = 9/11 * 8/10

In [25]:
P2

0.6545454545454545

Указанные события - не зависимы друг от друга, поэтому совместная вероятность вычисляется перемножением 2х вероятностей:

In [26]:
P = P1*P2

In [27]:
P

0.30545454545454537

##### Ответ: вероятность того, что все мячи белые равна 0.30545454545454537

Решение 4.2 Какова вероятность того, что ровно два мяча белые?

Возможные варианты 2 белых мячей из выбранных 4 мячей, вытащенных из 2 ящиков: 

Вариант №1: 1 ящик - 0 белых, 2 ящик - 2 белых

In [28]:
# ящик 1
P11 = (int(factorial(3) / (factorial(2) * factorial(3 - 2)))) * (int(factorial(7) / (factorial(0) * factorial(7 - 0)))) / (int(factorial(10) / (factorial(2) * factorial(10 - 2))))

In [29]:
P11

0.06666666666666667

In [30]:
# ящик 2
P12 = (int(factorial(9) / (factorial(2) * factorial(9 - 2)))) * (int(factorial(2) / (factorial(0) * factorial(2 - 0)))) / (int(factorial(11) / (factorial(2) * factorial(11 - 2))))

In [31]:
P12

0.6545454545454545

In [32]:
# вероятность, вариант №1
P1 = P11 * P12

In [33]:
P1

0.04363636363636363

Вариант №2: 1 ящик - 1 белый, 2 ящик - 1 белый

In [34]:
# ящик 1
P21 = (int(factorial(3) / (factorial(1) * factorial(3 - 1)))) * (int(factorial(7) / (factorial(1) * factorial(7 - 1)))) / (int(factorial(10) / (factorial(2) * factorial(10 - 2))))

In [35]:
P21

0.4666666666666667

In [36]:
# ящик 2
P22 = (int(factorial(9) / (factorial(1) * factorial(9 - 1)))) * (int(factorial(2) / (factorial(1) * factorial(2 - 1)))) / (int(factorial(11) / (factorial(2) * factorial(11 - 2))))

In [37]:
P22

0.32727272727272727

In [38]:
# вероятность, вариант №2
P2 = P21 * P22

In [39]:
P2

0.15272727272727274

Вариант №3: 1 ящик - 2 белых, 2 ящик - 0 белых

In [40]:
# ящик 1
P31 = (int(factorial(3) / (factorial(0) * factorial(3 - 0)))) * (int(factorial(7) / (factorial(2) * factorial(7 - 2)))) / (int(factorial(10) / (factorial(2) * factorial(10 - 2))))

In [41]:
P31

0.4666666666666667

In [42]:
# ящик 2
P32 = (int(factorial(9) / (factorial(0) * factorial(9 - 0)))) * (int(factorial(2) / (factorial(2) * factorial(2 - 2)))) / (int(factorial(11) / (factorial(2) * factorial(11 - 2))))

In [43]:
P32

0.01818181818181818

In [44]:
# вероятность, вариант №3
P3 = P31 * P32

In [45]:
P3

0.008484848484848484

Считаем совместную вероятность:

In [46]:
P = P1 + P2 + P3

In [47]:
P

0.20484848484848486

##### Ответ: вероятность того, что ровно два мяча белые равна 0.20484848484848486

Решение 4.3 Какова вероятность того, что хотя бы один мяч белый?

Посчитаем, что вероятность того, что все выбранные мячи НЕ белые:

In [48]:
# ящик 1
P1 = 3/10 * 2/9
# ящик 2
P2 = 2/11 * 1/10
# совместная вероятность
P = P1*P2

In [49]:
P

0.0012121212121212121

Вероятность того, что хотя бы один мяч белый обратна вероятности того, что все выбранные мячи НЕ белые:

In [50]:
P1 = 1 - P

In [51]:
P1

0.9987878787878788

##### Ответ: вероятность того, что хотя бы один мяч белый равна 0.9987878787878788